File: goormlevel/43068/1. 인공지능 청소기/인공지능 청소기.py
import sys

input = sys.stdin.readline
write = sys.stdout.write

def solve():
	n = int(input().rstrip())

	for _ in range(n):
		x, y, sec = map(int, input().split())

		if x < 0: x = -x
		if y < 0: y = -y

		if (x + y) <= sec and (sec - (x + y)) % 2 == 0:
			write("YES\n")
		else:
			write("NO\n")

if __name__ == "__main__":
	solve()

File: goormlevel/43068/1. 인공지능 청소기/인공지능 청소기.c
#include <iostream>

using namespace std;

int main() {
	int n = 0;
	if (!(cin >> n)) return 0;

	while (n--) {
		int x = 0, y = 0, sec = 0;
		if (!(cin >> x >> y >> sec)) return 0;

		if (x < 0) x = -x;
		if (y < 0) y = -y;

		if ((x + y) <= sec && (sec - (x + y)) % 2 == 0) {
			cout << "YES\n";
		}
		else {
			cout << "NO\n";
		}
	}

	return 0;
}